engine/render/src/waf_render.py

```diff
@@ -20,12 +20,59 @@
 def configure(conf):
     pass
 
-def transform_material(task, msg):
-    msg.vertex_program = msg.vertex_program.replace('.vp', '.vpc')
-    msg.fragment_program = msg.fragment_program.replace('.fp', '.fpc')
-    return msg
+waflib.Task.task_factory('material', '${JAVA} -classpath ${CLASSPATH} com.dynamo.bob.pipeline.MaterialBuilder ${SRC} ${TGT} ${SHADER_NAME} ${CONTENT_ROOT}',
+                      color='PINK',
+                      after='proto_gen_py',
+                      before='c cxx',
+                      shell=False)
 
-proto_compile_task('material', 'render.material_ddf_pb2', 'material_ddf_pb2.MaterialDesc', '.material', '.materialc', transform_material)
+waflib.Task.task_factory('material_shaderbuilder', '${JAVA} -classpath ${CLASSPATH} com.dynamo.bob.pipeline.ShaderProgramBuilder ${FP} ${VP} ${TGT} ${PLATFORM}',
```

Comment:
This change will introduce some latency to building since we can't do just a text replacement like before. But on the other hand, this means that we ACTUALLY use the pipeline to build content. 

A future improvement could perhaps be to gather all files that should be built, their output paths and potential options and pass that to bob to just churn through from the same instance instead of invoking bob for every resource like this. It's really slow to boot up bob, but the actual crosscompilation is really fast.